### ISSUE TITLE
Can't set icon in the new Window APIs

### ISSUE THREAD

**Issue by kirill-grouchnikov** [JetBrains/compose-jb#949]
```
fun main() = application {
    val state = rememberWindowState(
        placement = WindowPlacement.Floating,
        position = WindowPosition.Aligned(Alignment.Center),
        size = WindowSize(300.dp, 200.dp)
    )

    Window(
        title = "Demo",
        icon = ColorPainter(Color.Red),
        state = state,
        onCloseRequest = ::exitApplication,
    ) {
        Box(modifier = Modifier.background(Color.Blue).fillMaxSize())
    }
}
```

This crashes under build 270 with:

```
Exception in thread "main" java.lang.IllegalArgumentException: Cannot convert Painter with unspecified size. Please set size explicitly.
	at androidx.compose.ui.graphics.DesktopImageConverters_desktopKt.asAwtImage-Ug5Nnss(DesktopImageConverters.desktop.kt:78)
	at androidx.compose.ui.graphics.DesktopImageConverters_desktopKt.asAwtImage-Ug5Nnss$default(DesktopImageConverters.desktop.kt:73)
	at androidx.compose.ui.util.Windows_desktopKt.setIcon(Windows.desktop.kt:149)
	at androidx.compose.ui.window.Window_desktopKt$Window$5$1$2.invoke(Window.desktop.kt:171)
	at androidx.compose.ui.window.Window_desktopKt$Window$5$1$2.invoke(Window.desktop.kt:171)
	at androidx.compose.ui.util.ComponentUpdater$UpdateScope.set(ComponentUpdater.kt:45)
	at androidx.compose.ui.window.Window_desktopKt$Window$5$1.invoke(Window.desktop.kt:171)
	at androidx.compose.ui.window.Window_desktopKt$Window$5$1.invoke(Window.desktop.kt:169)
	at androidx.compose.ui.util.ComponentUpdater.update(ComponentUpdater.kt:27)
	at androidx.compose.ui.window.Window_desktopKt$Window$5.invoke(Window.desktop.kt:169)
	at androidx.compose.ui.window.Window_desktopKt$Window$5.invoke(Window.desktop.kt:168)
	at androidx.compose.ui.window.AwtWindow_desktopKt$AwtWindow$3.invoke(AwtWindow.desktop.kt:88)
	at androidx.compose.ui.window.AwtWindow_desktopKt$AwtWindow$3.invoke(AwtWindow.desktop.kt:87)
	at androidx.compose.ui.util.UpdateEffect_desktopKt$UpdateEffect$2$performUpdate$2.invoke(UpdateEffect.desktop.kt:58)
	at androidx.compose.ui.util.UpdateEffect_desktopKt$UpdateEffect$2$performUpdate$2.invoke(UpdateEffect.desktop.kt:57)
	at androidx.compose.runtime.snapshots.Snapshot$Companion.observe(Snapshot.kt:1776)
	at androidx.compose.runtime.snapshots.SnapshotStateObserver.observeReads(SnapshotStateObserver.kt:123)
	at androidx.compose.ui.util.UpdateEffect_desktopKt$UpdateEffect$2.invoke$performUpdate(UpdateEffect.desktop.kt:54)
	at androidx.compose.ui.util.UpdateEffect_desktopKt$UpdateEffect$2.invoke(UpdateEffect.desktop.kt:62)
	at androidx.compose.ui.util.UpdateEffect_desktopKt$UpdateEffect$2.invoke(UpdateEffect.desktop.kt:47)
	at androidx.compose.runtime.DisposableEffectImpl.onRemembered(Effects.kt:81)
	at androidx.compose.runtime.CompositionImpl$RememberEventDispatcher.dispatchRememberObservers(Composition.kt:780)
	at androidx.compose.runtime.CompositionImpl.applyChanges(Composition.kt:638)
	at androidx.compose.runtime.Recomposer.composeInitial$runtime(Recomposer.kt:733)
	at androidx.compose.runtime.CompositionImpl.setContent(Composition.kt:432)
	at androidx.compose.ui.window.Application_desktopKt$awaitApplication$2$1$2.invokeSuspend(Application.desktop.kt:176)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

Under the hood:

```
internal fun Window.setIcon(painter: Painter?) {
    setIconImage(painter?.asAwtImage(density, layoutDirection))
}
```
This calls `java.awt.Window.setIconImage` with the result of the conversion from Compose's `Painter` to AWT's `Image`. Except that:

```
fun Painter.asAwtImage(
    density: Density,
    layoutDirection: LayoutDirection,
    size: Size = intrinsicSize
): Image {
    require(size.isSpecified) {
        "Cannot convert Painter with unspecified size. Please set size explicitly."
    }
    ...
```

It expects the size to be specified

**Comment by igordmn** [comment]
Thanks!

Fixed in 1.0.0-alpha1-rc1

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-6347">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
